src/native-stack/views/NativeStackView.tsx

```diff
@@ -93,12 +109,31 @@ const MaybeNestedStack = ({
     </Container>
   );
 
+  const insets = useSafeAreaInsets();
+  const dimensions = useSafeAreaFrame();
+  // landscape is meaningful only for iPhone
+  const isLandscape =
+    dimensions.width > dimensions.height &&
+    !(Platform as PlatformIOSStatic).isPad &&
+    !(Platform as PlatformIOSStatic).isTVOS;
+  // `modal` and `formSheet` presentations do not take whole screen, so should not take the inset.
+  const isFullScreenModal =
+    stackPresentation !== 'modal' && stackPresentation !== 'formSheet';
+  const topInset = isFullScreenModal && !isLandscape ? insets.top : 0;
```

Comment:
These variables could potentially be moved into the `getDefaultHeaderHeight` function

Reply:
I agree with @kacperkapusciak. Could you move them to `getDefaultHeaderHeight` or is there a reason for them being here?